operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -718,9 +720,18 @@ private Map<String, Object> buildKafkaConfig(ManagedKafka managedKafka, Kafka cu
         config.put("kas.policy.shared-admin.adminclient-listener.protocol", "SSL");
 
         if (cruiseControlEnabled) {
-            config.put("cruise.control.metrics.topic.min.insync.replicas", this.configs.getConfig(managedKafka).cruiseControl.getMetricReporterTopicMinInsyncReplicas());
+            config.put("cruise.control.metrics.topic.min.insync.replicas", instanceConfig.cruiseControl.getMetricReporterTopicMinInsyncReplicas());
         }
 
+        Quantity cpu = new Quantity(kafkaConfigs.getContainerCpu());
+        BigDecimal cpuBytes = Quantity.getAmountInBytes(cpu);
+        // cpuBytes might be less than 1
+        int cpuCores = Math.max(1, cpuBytes.intValue());
+
+        int volumeCount = kafkaConfigs.getVolumeCount();
+        // since the thread number is per data dir, we should consider the volume count in the broker
+        config.put("num.recovery.threads.per.data.dir", cpuCores / volumeCount);
```

Comment:
I've always subscribed to the rule of thumb when doing IO heavy workloads (mostly maven builds to be fair) of using two threads per core.  So I'm tempted to say we should set `Math.floor(2 * (cpuCores / volumeCount))`.

The other thought that goes through my head here, is this formula assumes each volume is equally important & equally large (needing an equal share of the resources). When ever I've considered multiple volume support in RHOSAK I've been thinking about a customer data Dir and a service data dir. I appreciate that giving 80% of the cores to the customer data dir and 20% to the service is not an option given the Kafka API I wonder if its better to just set the value to core count and let the additional threads for the service data dir idle out?

Reply:
Thanks for the suggestion. I'll stick to 1 thread per core, unless we agree that's what we expected. And also revert the "partial" multiple volume support.